### ISSUE TITLE
twilio video camera switching issue node js webpack

### ISSUE THREAD

**Issue by harwinder-developer** [twilio/twilio-video.js#590]
I am facing the issue in camera switching 
my issue is matching with this thread https://github.com/twilio/twilio-video.js/issues/327 i have tried the solution explained in the above thread but it is not helping me at all.

The problem is when i switch my iphone camera then it works first time. then if i again switch to front it works , after that when i switch to back again it shows blank screen at participant side.
i am using the following code 

CODE FOR SWITCHING CAMERA
```
function updateVideoDevice(deviceId, room, mode) {
  const localParticipant = room.localParticipant;
  if (deviceId !== '') {
    console.log(deviceId);
    Video.createLocalVideoTrack({
      deviceId: { exact: deviceId }
    }).then(function (localVideoTrack) {
      const tracks = Array.from(localParticipant.videoTracks.values());
      localParticipant.unpublishTracks(tracks);
      detachTracks(tracks);
      localParticipant.publishTrack(localVideoTrack);
      const previewContainer = document.getElementById('talksdk-local-media');
      previewContainer.innerHTML = '';
      attachTracks([localVideoTrack], previewContainer);
    }).catch(function (error) {
    });
  }
}
```
here is the function attach track and detach track used in above function 
```
function detachTracks(tracks) {
  tracks.forEach(track => {
    // track.unpublish();
    track.detach().forEach(detachedElement => {
      detachedElement.remove();
    });
  });
}
function attachTracks(tracks, container) {
  tracks.forEach(track => {
    container.appendChild(track.attach());
  });
}>
```

This is the code for updating the remove video when a participant track is published or unpublished 

```
function trackSubscribed(div, track) {
    const videoElement1 = track.attach();
    if (track.kind === 'video') {
      videoElement1.id = 'participant-video';
    }
    div.appendChild(videoElement1);
  }
  function trackUnsubscribed(track) {
    track.detach().forEach(element => element.remove());
    $('#participant-video').remove();
  }
  function participantConnected(participant) {
    console.log('Participant "%s" connected', participant.identity);
    const participantMediaContainer = document.getElementById('talksdk-participant-media');
    participant.on('trackSubscribed', track => trackSubscribed(participantMediaContainer, track));
    participant.tracks.forEach(track => trackSubscribed(participantMediaContainer, track));
    participant.on('trackUnsubscribed', trackUnsubscribed);
  }
```
can you please check my code and let me know what information i can provide you more to fix this issue.

**Comment by manjeshbhargav** [comment]
Ho @harwinder-developer ,

Thanks for writing in with your issue. It looks like you are not stopping the LocalVideoTrack that you are unpublishing. So, the device will still be attached to this LocalVideoTrack. Try calling `LocalVideoTrack.stop()` after unpublishing and let me know if that works.

Thanks,

Manjesh Malavalli
JSDK Team

**Comment by harwinder-developer** [comment]
Thanks Manjesh, for your reply.
i have update my library to beta version and it is working fine now , also seems it is fast as compare to another library. 
now i am facing a issue while disconnecting a room . it remove the video tag from DOM but it is not turning off the camera light . this is the code which is present for beta version , let me know what thing i am missing in it.

```
room.on('disconnected', room => {
   room.localParticipant.tracks.forEach(publication => {
        const attachedElements = publication.track.detach();
        publication.track.stop();
        room.localParticipant.unpublishTrack(publication.track);
        attachedElements.forEach(element => element.remove());
      });
    });
```

Thanks!

**Comment by virgilioneves** [comment]
I am facing same issue with Safari 12.2 on iOS 12.2 (iPhone). In short:

(Step 1) Participant #1 (Chrome 74 on MacOS Mojave 10.14.3) joins the room - OK
(Step 2) Participant #2 (Safari 12.2 on iOS 12.2) joins the room with front camera - OK
(Step 3) Participant #2 attempts to switch to back camera - FAIL

At participant #2, everything seems to work fine after unpublishing old front camera track and publishing new back camera track. 

But at participant #1, only "trackUnsubscribed" events get fired properly. While I can see at twilio´s  log some evidences from the new video track being published, the associated event "trackSubscribed" does not get fired 

twilio-video.js:17396 2019-05-24 22:20:24.918Z | DEBUG in [RemoteParticipant #2: PAe507d0e5c54fd6433f125f0e1012e5cb]: VideoTrackPublication: RemoteVideoTrackPublication {trackName: "dd59db65-a60f-436a-af76-2bfc858e4b52", trackSid: "MT400ebc6e0eb454a1aca79bb27d9d9302", kind: "video", …}

IMPORTANT: Everything works fine with participant #2 using Chrome on MacOS and Android
